ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/SchemaInfo.java

```diff
@@ -105,6 +128,21 @@ public int hashCode() {
                   .stream()
                   .map(f -> f.getName() + " " + f.getSchema().toTypeString())
                   .collect(Collectors.joining(", ", SqlBaseType.STRUCT + "<", ">")))
+          .put(
+              SqlBaseType.DECIMAL,
+              si -> {
+                final Object precision = si.getFieldProperties().get("precision");
+                final Object scale = si.getFieldProperties().get("scale");
+                String parameterString = "";
+                if (precision != null) {
+                  parameterString += precision.toString();
+                  parameterString += ", ";
+                }
+                if (scale != null) {
+                  parameterString += scale.toString();
+                }
+                return SqlBaseType.DECIMAL + "(" + parameterString  + ")";
```

Comment:
You asked for java-isms and you shall receive: 
```java
KsqlPreconditions.checkArgument(
  !(precision == null ^ scale == null), 
  "Unexpected Decimal SchemaInfo with only one of precision and scale: " + new ObjectMapper().writeValueAsString(si)
);

final String precisionString = (scale != null) 
  ? String.format("(%s, %s)", precision, scale) 
  : "";
```
Note that `%s` will format anything as a string.